### PR TITLE
Invite model update

### PR DESCRIPTION
**Pull request by linusarnejohansson**
Updated the userwidgets/model and made sure that the api can handle reinvites and the new changeable.Invite type.

this PR is backwards compatible so if it receives an email it doesnt recognize then it is still going to send out an invite.

### CODE REVIEW

File: Context/Applications/Organizations.ts [hunk 1]
```diff
@@ -54,9 +54,13 @@ export class Organizations {
 			else {
 				const removed = current.users.filter(user => !response.users.includes(user))
 				const added = response.users.filter(user => !current.users.includes(user))
+				// needInvite = added (maybe some of them) + reinvites (all)
+				const needInvite = organization.users?.filter(userwidgets.Organization.Changeable.Invite.is)
+				// sendInvitesTo = added (all) + reinvites (all)
+				const sendInvitesTo = [...new Set([...added, ...(needInvite?.map(({ user }) => user) ?? [])])]
```
Comment (linusarnejohansson) [inline Context/Applications/Organizations.ts:60]: im not happy wih the name sendInvitesTo. 

Any suggestions?

File: Context/Applications/Organizations.ts [hunk 2]
```diff
@@ -54,9 +54,13 @@ export class Organizations {
 			else {
 				const removed = current.users.filter(user => !response.users.includes(user))
 				const added = response.users.filter(user => !current.users.includes(user))
+				// needInvite = added (maybe some of them) + reinvites (all)
+				const needInvite = organization.users?.filter(userwidgets.Organization.Changeable.Invite.is)
```
Comment (linusarnejohansson) [inline Context/Applications/Organizations.ts:58]: Im not happy with the name needInvite.

Any suggestions?

Comment (TheLillyleaf) [reply Context/Applications/Organizations.ts:58]: usersRequiringInvites?

File: Context/Applications/Organizations.ts (hunk 1 above)
Comment (TheLillyleaf) [reply Context/Applications/Organizations.ts:60]: usersToBeInvited?

File: Context/Applications/Organizations.ts [hunk 3]
```diff
@@ -54,9 +54,13 @@ export class Organizations {
...
+				// sendInvitesTo = added (all) + reinvites (all)
```
Comment (TheLillyleaf) [inline Context/Applications/Organizations.ts:59]: remove

File: Context/Applications/Organizations.ts [hunk 4]
```diff
@@ -54,9 +54,13 @@ export class Organizations {
 			else {
 				const removed = current.users.filter(user => !response.users.includes(user))
 				const added = response.users.filter(user => !current.users.includes(user))
+				// needInvite = added (maybe some of them) + reinvites (all)
```
Comment (TheLillyleaf) [inline Context/Applications/Organizations.ts:57]: remove

**Review by TheLillyleaf — APPROVED** [review]
Make changes if you want, but I think a more descriptive name can be better. Can't think of any shorter name tbh.

**Pull request merged**